Team417_PowerPlay/src/main/java/org/firstinspires/ftc/team417_PowerPlay/BaseAutonomous.java

```diff
@@ -0,0 +1,147 @@
+                }
+            }
+
+            if (tagFound) {
+                telemetry.addLine("Tag of interest is in sight!\n\nLocation data:");
+                tagToTelemetry(tagOfInterest);
+            } else {
+                telemetry.addLine("Don't see tag of interest :(");
+
+                if (tagOfInterest == null) {
+                    telemetry.addLine("(The tag has never been seen)");
+                } else {
+                    telemetry.addLine("\nBut we HAVE seen the tag before; last seen at:");
+                    tagToTelemetry(tagOfInterest);
+                }
+            }
+        }
+        telemetry.update();
+        sleep(20);
+    }
+
+    public void updateTelemetryAfterStart() {
+        if (tagOfInterest != null) {
+            telemetry.addLine("Tag snapshot:\n");
+            tagToTelemetry(tagOfInterest);
+            telemetry.update();
+        } else {
+            telemetry.addLine("No tag snapshot available, it was never sighted during the init loop :(");
+            telemetry.update();
+        }
+    }
+
+    void tagToTelemetry(AprilTagDetection detection) {
+        telemetry.addLine(String.format("\nDetected tag ID=%d", detection.id));
+        telemetry.addLine(String.format("Translation X: %.2f feet", detection.pose.x*FEET_PER_METER));
+        telemetry.addLine(String.format("Translation Y: %.2f feet", detection.pose.y*FEET_PER_METER));
+        telemetry.addLine(String.format("Translation Z: %.2f feet", detection.pose.z*FEET_PER_METER));
+        telemetry.addLine(String.format("Rotation Yaw: %.2f degrees", Math.toDegrees(detection.pose.yaw)));
+        telemetry.addLine(String.format("Rotation Pitch: %.2f degrees", Math.toDegrees(detection.pose.pitch)));
+        telemetry.addLine(String.format("Rotation Roll: %.2f degrees", Math.toDegrees(detection.pose.roll)));
+    }
+
+    public void raiseAndHoldArmGroundJunctionPosition() {
+        motorArm.setPower(0);
+        while (Math.abs(motorArm.getCurrentPosition() - GRD_JUNCT_ARM_POSITION) > 10 && opModeIsActive()) {
+            motorArm.setPower((GRD_JUNCT_ARM_POSITION - motorArm.getCurrentPosition()) * ARM_RAISE_POWER);
+        }
+        motorArm.setPower(HOLD_ARM_AT_GRD_POS_POWER);
+    }
+
+    public void raiseAndHoldArmMiddleJunctionPosition() {
+        motorArm.setPower(0);
+        while ((Math.abs(motorArm.getCurrentPosition() - (MID_JUNCT_ARM_POSITION)) > 10) && opModeIsActive()) {
+            motorArm.setPower((MID_JUNCT_ARM_POSITION - motorArm.getCurrentPosition()) * ARM_RAISE_POWER);
+
+        }
+        motorArm.setPower(HOLD_ARM_AT_MID_POS_POWER);
+    }
+
+    public void raiseAndHoldArmLowJunctionPosition() {
+        motorArm.setPower(0);
+        while (Math.abs(motorArm.getCurrentPosition() - LOW_JUNCT_ARM_POSITION) > 10) {
+            motorArm.setPower((LOW_JUNCT_ARM_POSITION - motorArm.getCurrentPosition()) * ARM_RAISE_POWER);
+        }
+        motorArm.setPower(HOLD_ARM_AT_MID_POS_POWER);
```

Comment:
Just because the only position that needed a higher power was the ground junction position because it's so low, I updated the variable name to be both for low and middle junction though!